macOS/dyld_cache_map_parser.cpp

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2022 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+https ://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "macOS/dyld_cache_map_parser.h"
+
+#include <iostream>
+#include <fstream>
+#include <regex>
+
+#include "common.h"
+
+struct Module {
+  Module(std::string n, uint64_t s, uint64_t e)
+  : name(n), start(s), end(e) {}
+  std::string name;
+  uint64_t start;
+  uint64_t end;
+};
+
+static bool is_page_aligned(uint64_t addr, uint64_t page_size = 0x4000) {
+  return !(addr & (page_size-1));
+}
+
+std::map<std::string, std::vector<std::string>> parse_dyld_map_file(const std::string &path) {
+  std::ifstream cache_map(path);
```

Comment:
There should be some kind of fatal error if the path doesn't exist (this stuff could change on the future macOS versions).

Reply:
yes, done.